baseplate/secrets/store.py

```diff
@@ -142,6 +160,53 @@ def get_raw(self, path):
         except KeyError:
             raise SecretNotFoundError(path)
 
+    def get_credentials(self, path):
+        """Decode and return a credential secret.
+
+        Credential secrets are a convention of username/password pairs stored as
+        separate values in the raw secret payload.
+
+        The following keys are significant:
+
+        ``type``
+            This must always be ``credential`` for this method.
+        ``encoding``
+            This must be un-set or set to ``identity``.
+        ``username``
+            This contains the raw username.
+        ``password``
+            This contains the raw password.
+
+        :rtype: :py:class:`CredentialSecret`
+
+        """
+        secret_attributes = self.get_raw(path)
+
+        if secret_attributes.get("type") != "credential":
+            raise CorruptSecretError(path, "secret does not have type=credential")
+
+        encoding = secret_attributes.get("encoding", "identity")
+
+        if encoding != "identity":
+            raise CorruptSecretError(path, "secret has encoding=%s rather than encoding=identity" % encoding)
+
+        missing_keys = []
+        values = {}
+        for key in ("username", "password"):
+            try:
+                values[key] = str(secret_attributes[key])
```

Comment:
I wasn't 100% sure what the best way to handle this was.  I don't feel like it makes sense to turn these values into bytes like we do for other secrets since these are text by their nature.  It seemed wrong to put the `str`-ing into `_decode_secret` since that is expected to return `bytes` so it seemed like the thing to do is just `str` the value directly here.

Reply:
Isn't it already a (unicode) string? This value is coming from `json.load` right?

Reply:
I imagine it could _technically_ be an int/float/boolean as well?  Although maybe not from vault.

Reply:
That sounds like a `CorruptSecretError` to me!